src/lib/api/upload/file_tools.ts

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2019 by Filestack.
+ * Some rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { File as FsFile } from './file';
+import fileType from 'file-type';
+import issvg from 'is-svg';
+import * as isutf8 from 'isutf8';
+import { isNode, sanitizeName } from './../../utils';
+import { FilestackError } from './../../../FilestackError';
+
+export type RawFile = Blob | Buffer | File | string;
+export type NamedInputFile = {
+  name?: string;
+  file: RawFile;
+};
+
+export type InputFile = RawFile | NamedInputFile;
+
+const base64Regexp = /data:([a-zA-Z]*\/[a-zA-Z]*);base64,([^\"]*)/i;
+
+/**
+ * Check if file is buffer
+ *
+ * @node
+ * @param input
+ */
+const isFileBuffer = (input: InputFile): input is Buffer => {
+  return Buffer.isBuffer(input);
+};
+
+/**
+ * Check if file is blob
+ * @param input
+ */
+const isFileBlob = (input: InputFile): input is Blob => {
+  return  input.toString() === '[object Blob]';
```

Comment:
Maybe something more like this one https://stackoverflow.com/a/20727783/5056566 :)

Reply:
Nope it will not work because of: 
```
 const a = new File([1,2,3], "foo.txt", {type: "text/plain"});
const b = new Blob([1,2,3]);
08:41:41.122 undefined
08:41:48.371 a instanceof Blob
08:41:48.378 true
08:41:52.769 b instanceof Blob
08:41:52.776 true
08:42:02.450 a.toString();
08:42:02.486 "[object File]"
08:42:07.056 b.toString();
08:42:07.095 "[object Blob]"
```